From error27@gmail.com  Thu Aug 19 10:34:59 2010
Date: Wed, 11 Aug 2010 20:01:46 +0200
From: Dan Carpenter <error27@gmail.com>
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Paul Fulghum <paulkf@microgate.com>,
	Alan Cox <alan@linux.intel.com>,
	Stephen Hemminger <shemminger@vyatta.com>,
	kernel-janitors@vger.kernel.org
Subject: synclink: add mutex_unlock() on error path
Message-ID: <20100811180146.GB645@bicker>
Content-Disposition: inline

There is a path which still holds its mutex here.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Acked-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/synclink_gt.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/char/synclink_gt.c
+++ b/drivers/char/synclink_gt.c
@@ -691,8 +691,10 @@ static int open(struct tty_struct *tty,
 	if (info->port.count == 1) {
 		/* 1st open on this device, init hardware */
 		retval = startup(info);
-		if (retval < 0)
+		if (retval < 0) {
+			mutex_unlock(&info->port.mutex);
 			goto cleanup;
+		}
 	}
 	mutex_unlock(&info->port.mutex);
 	retval = block_til_ready(tty, filp, info);
